cmd/injective-price-oracle/oracle.go

```diff
@@ -228,3 +294,34 @@ func oracleCmd(cmd *cli.Cmd) {
 		closer.Hold()
 	}
 }
+
+func ConnectWebSocket(ctx context.Context, websocketUrl, urlHeader string) (conn *websocket.Conn, err error) {
+	u, err := url.Parse(websocketUrl)
+	if err != nil {
+		log.Fatal("Error parsing URL:", err)
+		return &websocket.Conn{}, err
+	}
+
+	header := http.Header{}
+	header.Add("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(urlHeader)))
+
+	dialer := websocket.DefaultDialer
+	dialer.EnableCompression = true
+	retries := 0
+	for {
+		conn, _, err = websocket.DefaultDialer.DialContext(ctx, u.String(), header)
+		if err != nil {
+			log.Infof("Failed to connect to WebSocket server: %v", err)
+			retries++
+			if retries > oracle.MaxRetriesReConnectWebSocket {
+				log.Infof("Reached maximum retries (%d), exiting...", oracle.MaxRetriesReConnectWebSocket)
+				return
+			}
+			log.Infof("Retrying connect %sth in 5s...", fmt.Sprint(retries))
+			time.Sleep(5 * time.Second)
```

Comment:
This code is blocking, it should acknowledge the context
```suggestion
			select {
			case <-ctx.Done():
				return nil, ctx.Err()
			case <-time.NewTimer(5*time.Second).C:
			}
```